vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc

```diff
@@ -1055,4 +1055,22 @@ bool ConfirmationsImpl::IsRetryingToGetRefillSignedTokens() const {
   return true;
 }
 
+void ConfirmationsImpl::ResetConfirmationsState(
+    OnResetConfirmationsStateCallback callback) {
+  is_initialized_ = false;
+  state_has_loaded_ = false;
+  confirmations_client_->ResetState(_confirmations_name,
+      std::bind(&ConfirmationsImpl::OnResetConfirmationsState,
+        this, callback, _1));
+}
+
+void ConfirmationsImpl::OnResetConfirmationsState(
+    OnResetConfirmationsStateCallback callback,
+    const ledger::Result result) {
+  StopRetryingToGetRefillSignedTokens();
```

Comment:
this seems very fragile. We need a more generalized Shutdown method for all of Ledger to stop anything currently in progress and that should be called before deleting the state

Reply:
cc @tmancey is there a better way?